### PR TITLE
Add st7701s and rpi_dpi_rgb docs

### PR DESCRIPTION
**Pull request by clydebarrow**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#5872

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by DannyN2222** [comment]
I dont think i can raise a PR to a PR so ill just add it as a comment
this display (https://www.elecrow.com/esp32-display-7-inch-hmi-display-rgb-tft-lcd-touch-screen-support-lvgl.html) works with this example config:
```
esphome:
  name: 7.0_inch_800x480_rgb_tft_lcd_example_config
  platformio_options:
    board_build.flash_mode: dio
esp32:
  board: esp32-s3-devkitc-1
  flash_size: 4MB
  framework:
    type: esp-idf
# Required to achieve sufficient PSRAM bandwidth
    sdkconfig_options:
      CONFIG_ESP32S3_DEFAULT_CPU_FREQ_240: y
      CONFIG_ESP32S3_DATA_CACHE_64KB: y
      CONFIG_SPIRAM_FETCH_INSTRUCTIONS: y
      CONFIG_SPIRAM_RODATA: y
display:
  - platform: rpi_dpi_rgb
    id: main_display
    color_order: RGB
    auto_clear_enabled: false
    dimensions:
      width: 800
      height: 480
    de_pin: 41
    hsync_pin: 39
    vsync_pin: 40
    pclk_pin:
      number: 0
      ignore_strapping_warning: true
    pclk_frequency: 25MHz
    update_interval: 3s
    hsync_pulse_width: 48
    hsync_front_porch: 40
    hsync_back_porch: 40
    vsync_pulse_width: 31
    vsync_front_porch: 1
    vsync_back_porch: 40
    # Replace XX with the correct pin number
    # Order is for big_endian (typically required.)
    data_pins:
      - 8        #g3
      - 16        #g4
      - 1        #g5
      - 14        #r1
      - 21        #r2
      - 47        #r3
      - 48        #r4
      - number: 45        #r5
        ignore_strapping_warning: true
      - 15        #b1
      - 7        #b2
      - 6        #b3
      - 5        #b4
      - 4        #b5
      - 9        #g0
      - number: 46        #g1
        ignore_strapping_warning: true
      - number: 3        #g2
        ignore_strapping_warning: true
```

File: components/display/rpi_dpi_rgb.rst [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+- **rotation** (*Optional*): Rotate the display presentation in software. Choose one of ``0°``, ``90°``, ``180°``, or ``270°``.
+- **lambda** (*Optional*, :ref:`lambda <config-lambda>`): The lambda to use for rendering the content on the display.
+  See :ref:`display-engine` for more information.
+
+
+The horizontal and vertical ``pulse_width``, ``front_porch`` and ``back_porch`` values are optional, but may require
+changing for a specific display. Refer to the manufacturer's sample code for suitable values. These specify timing
+requirements for the display.
+
+# Example configurations
```
Comment (jesserockz) [inline components/display/rpi_dpi_rgb.rst]:
```suggestion
Example configurations
----------------------
```

File: components/display/rpi_dpi_rgb.rst [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+.. code-block:: yaml
+
+    # Waveshare ESP32-S3 Touch 4.3 config
```
Comment (jesserockz) [inline components/display/rpi_dpi_rgb.rst]:
```suggestion

Waveshare ESP32-S3 Touch 4.3
****************************

.. code-block:: yaml

```

File: components/display/rpi_dpi_rgb.rst [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+          green:
+            - 39        #g2
+            - 0         #g3
+            - 45        #g4
+            - 48        #g5
+            - 47        #g6
+            - 21        #g7
+
+
+    # Makerfabs 4.3" 800x480 display
```
Comment (jesserockz) [inline components/display/rpi_dpi_rgb.rst]:
```suggestion
Makerfabs 4.3" 800x480 display
******************************

.. code-block:: yaml

```

File: components/display/st7701s.rst [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+sequence (the default and currently the only sequence is 1), the remainder must be byte arrays providing additional
+init commands, each consisting of a command byte followed by zero or more data bytes.
+
+These will be collected and sent to the display via SPI during initialisation.
+
+# Example configurations
+
+.. code-block:: yaml
+
+    # Seeed Sensecap Indicator configuration.
```
Comment (jesserockz) [inline components/display/st7701s.rst]:
```suggestion
Example configurations
----------------------

Seeed Sensecap Indicator
************************

.. code-block:: yaml

```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**